src/output-json.c

```diff
@@ -438,8 +444,69 @@ static void EveAddMetadata(const Packet *p, const Flow *f, JsonBuilder *js)
 
 int CreateJSONEther(JsonBuilder *parent, const Packet *p, const MacSet *ms);
 
+/* Callback function to pack payload contents from a stream into a buffer
+ * so we can report them in JSON output. */
+static int JsonDumpStreamSegmentCallback(const Packet *p, void *data, const uint8_t *buf, uint32_t buflen)
+{
+    MemBuffer *payload = (MemBuffer *)data;
+    MemBufferWriteRaw(payload, buf, buflen);
+
+    return 1;
+}
+
+
+static void EveAddStreamDataOneWay(const Packet *p, const Flow *f, JsonBuilder *js,
+                                    uint8_t flags, MemBuffer *payload, uint8_t flag)
```

Comment:
why use the MemBuffer here? Can't we just stream the data right into the `js`?

Reply:
I did do that because the stream is assembled by chunk and then added to the js. I did look at json builder and did not find  a way to concatenate string on a field.